File: backend/app/models/task.py
from sqlalchemy import Column, String, DateTime, ForeignKey, Integer, func
from sqlalchemy.orm import relationship
from app.core.database import Base

class TeachingTask(Base):
    __tablename__ = "teaching_tasks"

    id = Column(String, primary_key=True, index=True)
    teacher_id = Column(String, ForeignKey("teachers.id", ondelete="CASCADE"), nullable=False)
    class_id = Column(String, ForeignKey("classes.id", ondelete="CASCADE"), nullable=False)
    subject = Column(String, nullable=False)
    topic = Column(String, nullable=False)
    duration_minutes = Column(Integer, nullable=False)
    language = Column(String, nullable=False)  # "mr", "hi", "en"
    status = Column(String, nullable=False, default="draft")
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now(), server_default=func.now())

    teacher = relationship("Teacher")
    class_ = relationship("Class")

File: frontend/src/types/task.ts
export interface Teacher {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
}

export interface Class {
  id: string;
  teacher_id: string;
  name: string;
  grade: string;
  section: string;
  primary_language: string;
  created_at: string;
  updated_at: string;
}

export interface TeachingTask {
  id: string;
  teacher_id: string;
  class_id: string;
  subject: string;
  topic: string;
  duration_minutes: number;
  language: string;
  status: string;
  created_at: string;
  updated_at: string;
}
